### ISSUE TITLE
Error message on exit when using bgfx on Linux

### ISSUE THREAD

**Issue by cuavas** [mamedev/mame#5964]
MAME prints an error message on exit when using `-video bgfx` on Linux with OpenGL backend.  The message varies, but it's always a BadDrawable error.  Here are a couple of examples:
```
X Error of failed request:  BadDrawable (invalid Pixmap or Window parameter)
  Major opcode of failed request:  14 (X_GetGeometry)
  Resource id in failed request:  0x5e0000c
  Serial number of failed request:  1380
  Current serial number in output stream:  1380
```
```
X Error of failed request:  BadDrawable (invalid Pixmap or Window parameter)
  Major opcode of failed request:  72 (X_PutImage)
  Resource id in failed request:  0x5e0000c
  Serial number of failed request:  1838
  Current serial number in output stream:  1863
```
```
X Error of failed request:  BadDrawable (invalid Pixmap or Window parameter)
  Major opcode of failed request:  14 (X_GetGeometry)
  Resource id in failed request:  0x5e0000c
  Serial number of failed request:  1125
  Current serial number in output stream:  1125
```
* It doesn't happen if you exit from the system information screen - you have to let it get as far as displaying emulated video
* It happens with `-bgfx_screen_chains none`
* The message is printed to standard error (not standard output)
* It seems to have started happening after commit 4cccc0183cef3dc959500d708256d18a98a62c46
* It doesn't happen on Windows

I'm not particularly worried about this at the moment - it doesn't actually crash.  However it may indicate something being used after it's no longer valid.  If I get some time I might try to track down the call stack.  For now I'm just opening an issue so it can be tracked.

**Comment by couriersud** [comment]
I can not reproduce this issue on Ubuntu 16.04 with nvidia 440 drivers. Which commandline options and games did you test?

**Comment by cuavas** [comment]
It happens for every system I use.

**Comment by cuavas** [comment]
Fixed as of 3a7971b93fad66247edf3b422edb9b4ee7e5e3cb.
